cider/include/cider/processor/BatchProcessor.h

```diff
@@ -23,28 +23,32 @@
 #define CIDER_BATCH_PROCESSOR_H
 
 #include <memory>
-#include "BatchProcessorContext.h"
+
+#include "cider/processor/BatchProcessorContext.h"
 #include "cider/processor/JoinHashTableBuilder.h"
-#include "include/cider/batch/CiderBatch.h"
 #include "substrait/plan.pb.h"
 
-namespace cider::processor {
+struct ArrowArray;
+struct ArrowSchema;
+
+namespace cider::exec::processor {
 
 enum class BatchProcessorState {
   kRunning,
   kWaiting,
   kFinished,
 };
 
-class BatchProcessor {
+class BatchProcessor : public std::enable_shared_from_this<BatchProcessor> {
```

Comment:
Why need `std::enable_shared_from_this` since we use `BatchProcessor` in `Unique_ptr` way?

Reply:
when create the hash join handler it need a batchprocessor this shared_ptr reference